util/kubeconfig.go

```diff
@@ -0,0 +1,81 @@
+package util
+
+import (
+	"github.com/pkg/errors"
+	"net/http"
+	"os"
+	"strings"
+
+	"k8s.io/client-go/plugin/pkg/client/auth/exec"
+	restclient "k8s.io/client-go/rest"
+	"k8s.io/client-go/tools/clientcmd"
+	"k8s.io/client-go/transport"
+)
+
+
+type Config = restclient.Config
+
+func NewConfig() (config *Config) {
+	config, err := clientcmd.NewNonInteractiveDeferredLoadingClientConfig(
+		clientcmd.NewDefaultClientConfigLoadingRules(), &clientcmd.ConfigOverrides{}).ClientConfig()
+	if err != nil {
+		panic(err)
+	}
+
+	return
+}
+
+
+func GetBearerToken(in *restclient.Config, explicitKubeConfigPath string) (string, error) {
+
+	if len(in.BearerToken) > 0 {
+		return in.BearerToken, nil
+	}
+
+	if token := getEnvToken(); token != "" {
+		return token, nil
+	}
+
+	if in == nil {
+		return "", errors.Errorf("RestClient can't be nil")
+	}
+	if in.ExecProvider != nil {
+		tc, err := in.TransportConfig()
+		if err != nil {
+			return "", err
+		}
+
+		auth, err := exec.GetAuthenticator(in.ExecProvider)
+		if err != nil {
+			return "", err
+		}
+
+		//This function will return error because of TLS Cert missing,
+		// This code is not making actual request. We can ignore it.
+		_ = auth.UpdateTransportConfig(tc)
+
+		rt, err := transport.New(tc)
+		if err != nil {
+			return "", err
+		}
+		req := http.Request{Header: map[string][]string{}}
+
+		_, _ = rt.RoundTrip(&req)
+
+		token := req.Header.Get("Authorization")
+		return strings.TrimPrefix(token, "Bearer "), nil
+	}
+	if in.AuthProvider != nil {
+		if in.AuthProvider.Name == "gcp" {
+			token := in.AuthProvider.Config["access-token"]
+			return strings.TrimPrefix(token, "Bearer "), nil
+		}
+	}
+	return "", errors.Errorf("could not find a token")
+}
+
+
+// Get the Auth token from environment variable
```

Comment:
@aleksandrmelnikov this one is Argo specific, we don't need it.

Reply:
Removed